ps-modules/Build/Build.psm1

```diff
@@ -0,0 +1,265 @@
+       Invoke-Expression "$bootstrapScript"
+       Pop-Location
+   }
+}
+
+function Run-PreBuildScriptIfExists {
+   param(
+      $script
+   )
+   Run-ScriptIfExists -title "Pre-build step" -script $script
+}
+
+function Run-PostBuildScriptIfExists {
+   param(
+      $script,
+      $preStagePath
+   )
+   $scriptArgs = @{ BuildArtifactsPath = ((Resolve-Path $preStagePath).Path -replace '\\', '/') }
+   Run-ScriptIfExists -title "Post-build step" -script $script -scriptArgs $scriptArgs
+}
+
+function Install-Package
+{
+   param(
+      [string]$vcpkgExe,
+      [string]$package,
+      [PSObject]$triplets
+   )
+   Write-Banner -Level 3 -Title "Installing package: $package"
+   $tripletCount = $triplets.Length
+   Write-Message "Installing for $tripletCount triplet(s)..."
+
+   $tripletNum = 1
+   foreach ($triplet in $triplets) {
+      Write-Message "> Installing for triplet $tripletNum/$tripletCount`: $triplet..."
+      Install-FromVcPkg -Package $package -Triplet $triplet -vcpkgExe $vcpkgExe
+      Exit-IfError $LASTEXITCODE
+      $tripletNum++
+   }
+}
+
+function Create-FinalArtifacts {
+   param(
+      [string]$arm64Dir,
+      [string]$x64Dir,
+      [string]$preStagePath
+   )
+   if (-not (Get-isOnMacOS)) { return } # This is only required on Mac
+   Write-Banner -Level 3 -Title "Creating final Mac artifacts"
+   Create-FinalMacArtifacts -arm64Dir "$arm64Dir" -x64Dir "$x64Dir" -universalDir "$preStagePath"
+}
+
+function Stage-Artifacts {
+   param(
+      $vcpkgExe,
+      $preStagePath,
+      $stagePath,
+      $packageNameOnly,
+      $artifactName
+   )
+   
+   Write-Banner -Level 3 -Title "Stage build artifacts"
+
+   Write-Debug "Creating dir: $artifactName"
+   New-Item -Path $stagePath/$artifactName -ItemType Directory -Force -ErrorAction SilentlyContinue | Out-Null
+
+   $dependenciesFilename = "dependencies.json"
+   Write-Debug "Generating: `"$dependenciesFilename`"..."
+   Write-Debug "> $stagePath/$artifactName/$dependenciesFilename"
+   Invoke-Expression "$vcpkgExe list --x-json > $stagePath/$artifactName/$dependenciesFilename"
+
+   $packageInfoFilename = "package.json"
+   Write-Debug "Generating: `"$packageInfoFilename`"..."
+   $dependenciesJson = Get-Content -Raw -Path "$stagePath/$artifactName/$dependenciesFilename" | ConvertFrom-Json
+   $packageVersion = ($dependenciesJson.PSObject.Properties.Value | Where-Object { $_.package_name -eq $packageNameOnly } | Select-Object -First 1).version
+   Write-ReleaseInfoJson -PackageName $packageNameOnly -Version $packageVersion -PathToJsonFile "$stagePath/$artifactName/$packageInfoFilename"
+   
+   # TODO: Add info in this file on where each package was downloaded from
+   # TODO: Add license file info to the staged artifacts (ex. per-library LICENSE, COPYING, or other such files that commonly have license info in them)
+   
+   Write-Message "Copying: $preStagePath =`> $artifactName"
+   $excludedFolders = @("tools", "share", "debug")
```

Comment:
Eventually, we may want delegate this to PostBuild scripts so we have more control over what actually goes into the final artifact, but this is probably fine for now.

Reply:
I'm going to defer this for now.